school/src/test/java/ru/hogwarts/school/StudentControllerWebMvcTest.java

```diff
@@ -0,0 +1,207 @@
+
+                        .put("/student")
+                        .content(studentObject.toString())
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(id))
+                .andExpect(jsonPath("$.name").value(name))
+                .andExpect(jsonPath("$.age").value(age));
+        verify(studentRepository,times(1)).save(student);
+    }
+
+    @Test
+    public void findStudentTest() throws Exception {
+
+        final Long id = 1L;
+        final String name = "Harry";
+        final int age = 7;
+
+        JSONObject studentObject = new JSONObject();
+        studentObject.put("name", "Harry");
+        studentObject.put("age", 7);
+
+        Student student = new Student();
+        student.setId(id);
+        student.setName(name);
+        student.setAge(age);
+
+        when(studentRepository.findById(id)).thenReturn(Optional.of(student));
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .get("/student/" + id)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(id))
+                .andExpect(jsonPath("$.name").value(name))
+                .andExpect(jsonPath("$.age").value(age));
+
+    }
+    @Test
+    public void deleteStudentTest() throws Exception {
+
+        final Long id = 1L;
+        final String name = "Harry";
+        final int age = 7;
+
+        JSONObject studentObject = new JSONObject();
+        studentObject.put("name", "Harry");
+        studentObject.put("age", 7);
+
+        Student student = new Student();
+        student.setId(id);
+        student.setName(name);
+        student.setAge(age);
+
+        doNothing().when(studentRepository).deleteById(id);
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .delete("/student/" + id))
+                .andExpect(status().isOk());
+    }
+    @Test
+    public void findByAgeTest() throws Exception {
+        final Long id = 1L;
+        final String name = "Harry";
+        final int age = 7;
+
+        JSONObject studentObject = new JSONObject();
+        studentObject.put("name", "Harry");
+        studentObject.put("age", 7);
+
+        Student student = new Student();
+        student.setId(id);
+        student.setName(name);
+        student.setAge(age);
+
+        when(studentRepository.findByAge(age)).thenReturn(List.of(student));
+
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .get("/student")
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(id))
+                .andExpect(jsonPath("$.name").value(name))
+                .andExpect(jsonPath("$.age").value(age));
```

Comment:
```suggestion
        mockMvc.perform(MockMvcRequestBuilders
                        .get("/student?age="+age)
                        .accept(MediaType.APPLICATION_JSON))
                .andExpect(status().isOk())
                .andExpect(jsonPath("$[0].id").value(id))
                .andExpect(jsonPath("$[0].name").value(name))
                .andExpect(jsonPath("$[0].age").value(age));
```
Неправильно указываешь урл и достаёшь значения